indico/modules/events/sessions/util.py

```diff
@@ -59,3 +69,70 @@ def can_manage_sessions(user, event, role=None):
     """Check whether a user can manage any sessions in an event"""
     return event.can_manage(user) or any(s.can_manage(user, role)
                                          for s in event.sessions.options(joinedload('acl_entries')))
+
+
+def generate_csv_from_sessions(sessions):
+    """Generate a CSV file from a given session list.
+
+    :param sessions: The list of sessions to include in the file
+    """
+    column_names = ['ID', 'Title', 'Description', 'Code']
+    buf = BytesIO()
+    writer = csv.DictWriter(buf, fieldnames=column_names)
+    writer.writeheader()
+    for sess in sessions:
+        row = {'ID': sess.friendly_id, 'Title': sess.title.encode('utf-8'),
+               'Description': sess.description.encode('utf-8'), 'Code': sess.code.encode('utf-8')}
+        writer.writerow(row)
+    buf.seek(0)
+    return buf
+
+
+class SessionListToPDF(PDFBase):
+    def __init__(self, conf, sessions):
+        PDFBase.__init__(self, story=[])
+        self.conf = conf
+        self.sessions = sessions
+        self.PAGE_WIDTH, self.PAGE_HEIGHT = landscape(A4)
+
+    def getBody(self, story=None):
+        story = story or self._story
+        header_style = ParagraphStyle(name='header_style', fontSize=12, alignment=TA_CENTER)
+        story.append(Paragraph('<b>{}</b>'.format(_('List of sessions')), header_style))
+
+        text_style = ParagraphStyle(name='text_style', fontSize=8, alignment=TA_LEFT, leading=10, leftIndent=10)
+        text_style.fontName = 'Times-Roman'
+        text_style.spaceBefore = 0
+        text_style.spaceAfter = 0
+        text_style.firstLineIndent = 0
+
+        rows = []
+        row_values = []
+        for col in [_(col) for col in 'ID', 'Type', 'Title', 'Code', 'Description']:
+            row_values.append(Paragraph('<b>{}</b>'.format(col), text_style))
+        rows.append(row_values)
+
+        for sess in self.sessions:
+            rows.append([
+                Paragraph(sess.friendly_id, text_style),
+                Paragraph(_('Poster') if sess.is_poster else _('Standard'), text_style),
+                Paragraph(sess.title.encode('utf-8'), text_style),
+                Paragraph(sess.code.encode('utf-8'), text_style),
+                Paragraph(sess.description.encode('utf-8'), text_style)
+            ])
+
+        col_widths = (None,) * 5
+        table_style = TableStyle([
+            ('VALIGN', (0, 0), (-1, -1), 'MIDDLE'),
+            ('LINEBELOW', (0, 0), (-1, 0), 1, colors.black),
+            ('ALIGN', (0, 0), (-1, 0), 'CENTER'),
+            ('ALIGN', (0, 1), (-1, -1), 'LEFT')
+        ])
+        story.append(Table(rows, colWidths=col_widths, style=table_style))
+        return story
+
+
+def generate_pdf_from_sessions(conf, sessions):
```

Comment:
I would already pass an `Event` object here (called `event`). You can simply use `.as_legacy` to get the `Conference` from it. That way there's less to change in the future when we get rid of Conference